webapp/channels/src/plugins/registry.ts

```diff
@@ -1225,4 +1225,22 @@ export default class PluginRegistry {
 
         return id;
     });
+
+    // Register a schema for user settings. This will show in the user settings modals
+    // and all values will be stored in the preferences with cateogry pp_${pluginId} and
+    // the name of the setting.
+    //
+    // The settings definition can be found in /src/types/plugins/user_settings.ts
+    //
+    // Malformed settings will be filtered out.
+    registerUserSettings = reArg(['setting'], ({setting}) => {
```

Comment:
Nothing to be done in this PR, but I wonder if we'll want to put this in the `plugin.json` instead of requiring a plugin calls `registerUserSettings`. If we have it as JSON, then we could potentially show it on mobile as well, although I could also see cases where plugins want to modify this structure at runtime, so having this option would be good regardless.

Reply:
The main problem is the `onSubmit`. We could potentially set on the `onSubmit` a plugin route, and make an api call to the plugin directly (what most plugin will probably do either way).

That is the way we did it with apps, and yes, it could work, but 0/5 on going that direction.

Reply:
I guess that's only sometimes needed since, at least currently, you're not providing a custom `onSubmit` in your PR